src/main/scala/com/olegpy/meow/internal/HandleOptics.scala

```diff
@@ -0,0 +1,28 @@
+package com.olegpy.meow.internal
+
+import cats.mtl.ApplicativeHandle
+import com.olegpy.meow.optics.TPrism
+
+
+private[meow] object HandleOptics {
+  class Applicative[F[_], S, E](
+    parent: ApplicativeHandle[F, S],
+    prism: TPrism[S, E]
+  ) extends ApplicativeHandle[F, E] {
+    val applicative: cats.Applicative[F] = parent.applicative
+    def handle[A](fa: F[A])(f: E => A): F[A] =
+      parent.handle(fa) { case prism(e) => f(e) }
+
+    def handleWith[A](fa: F[A])(f: E => F[A]): F[A] =
+      parent.handleWith(fa) { case prism(e) => f(e) }
```

Comment:
It seems that here, `{ case prism(e) => f(e) }` is a partial function, so if we get a error `S` that is not `E`, this will raise a `MatchError` instead of doing a non-recovery. You need to have a fallback case and reraise the error (using `parent.raise`, not `this.raise`).

In [ApplicativeError](https://github.com/oleg-py/meow-mtl/blob/62ea5cd8acf88f7c6ba4ca4a87d6eb3f78318ffa/src/main/scala/com/olegpy/meow/internal/RaiseOptics.scala#L17) I used `recoverWith` - which is defined to take a `PartialFunction` specifically, to implement `handleErrorWith`, which takes a total one.

Would also really appreciate if you can add the test for that case. This won't be caught by generic laws, as they don't raise "S but not E" errors.

Reply:
Fixed and covered with a totality test.